setup.py

```diff
@@ -45,14 +45,21 @@
     try:
         if "Darwin" == platform.system():
             target: Optional[str] = os.environ.get("MACOSX_DEPLOYMENT_TARGET")
-            if None is target or float(target) < 10.15:
+            env_setup_needed: bool = False
+            if None is target:
+                env_setup_needed = True
+            else:
+                version_values: List[int] = [int(part) for part in target.split(".")]
+                if 10 >= version_values[0] and 15 >= version_values[1]:
+                    env_setup_needed = True
+            if env_setup_needed:
                 os.environ["MACOSX_DEPLOYMENT_TARGET"] = "10.15"
```

Comment:
How about something like:

```suggestion
            target_env_var_name = "MACOSX_DEPLOYMENT_TARGET"
            required_target = (10, 15)
            
            target_str: Optional[str] = os.environ.get(target_env_var_name)
            target: Tuple[int, ...] = ()
            if target_str is not None:
                target = tuple(int(part) for part in target_str.split("."))
            if target < required_target:
                target = required_target
            os.environ[target_env_var_name] = ".".join(str(part) for part in target)
```

That way we aren't repeating any constants (besides the version separator) and the version comparison is more Pythonic.

Reply:
Sure. Also added type annotations for `target_env_var_name` and `required_target `

Reply:
Maybe `required_target` should be `min_target`.

Reply:
Sure